### PR TITLE
[Reviewer: Andy] Migrate to etcd-based clustering

### PR DESCRIPTION
**Pull request by mirw**
Andy,

Please can you review my changes to use the etcd-based clustering function?  I've tested live.  I suggest we don't merge this until we cut the next release.

Thanks,

Matt

### CODE REVIEW

File: bono.yaml [hunk 1]
```diff
@@ -106,44 +111,24 @@ resources:
...
 
             # Now install the software.
-            DEBIAN_FRONTEND=noninteractive apt-get install bono dnsutils --yes --force-yes
+            DEBIAN_FRONTEND=noninteractive apt-get install clearwater-management bono --yes --force-yes
 
             # Update DNS
             retries=0
             while ! { nsupdate -y "__zone__:__dnssec_key__" -v << EOF
             server __dns_ip__
-            update delete bono.__zone__.
```
Comment (bossmc) [inline bono.yaml:146]: Seems like an unrelated fix, is this laying the groundwork for elastic scaling?

File: ellis.yaml [hunk 2]
```diff
@@ -108,6 +112,7 @@ resources:
             __dn_range_length__: { get_param: dn_range_length }
             __dns_ip__: { get_param: dns_ip }
             __dnssec_key__: { get_param: dnssec_key }
+            __etcd_ip__ : { get_param: etcd_ip }
```
Comment (bossmc) [inline ellis.yaml:140]: Surely this should be the `local_ip` if `etcd_ip` isn't specified?  Maybe I'm missing something...

File: ellis.yaml [hunk 3]
```diff
@@ -116,23 +121,30 @@ resources:
             curl -L http://repo.cw-ngv.com/repo_key | apt-key add -
             apt-get update
 
-            # Configure /etc/clearwater/config.
+            # Configure /etc/clearwater/local_config.
             mkdir -p /etc/clearwater
-            cat > /etc/clearwater/config << EOF
+            etcd_ip=__etcd_ip__
+            [ -n "$etcd_ip" ] || etcd_ip=$(hostname -I)
```
Comment (bossmc) [inline ellis.yaml:156]: Ah, ok here it is.

**Comment by bossmc** [comment]
Looks good, some rambling thought process in the comments but I don't think there's anything actionable here :grin:

File: bono.yaml (hunk 1 above)
Comment (mirw) [inline bono.yaml:146]: Good spot.  Yes, this is laying the groundwork for elastic scaling (or even just static scaling).
